src/transformers/models/deta/image_processing_deta.py

```diff
@@ -0,0 +1,1323 @@
+            prepared_annotations = []
+            for image, target in zip(images, annotations):
+                target = self.prepare_annotation(
+                    image, target, format, return_segmentation_masks=return_segmentation_masks, masks_path=masks_path
+                )
+                prepared_images.append(image)
+                prepared_annotations.append(target)
+            images = prepared_images
+            annotations = prepared_annotations
+            del prepared_images, prepared_annotations
+
+        # transformations
+        if do_resize:
+            if annotations is not None:
+                resized_images, resized_annotations = [], []
+                for image, target in zip(images, annotations):
+                    orig_size = get_image_size(image)
+                    resized_image = self.resize(image, size=size, max_size=max_size, resample=resample)
+                    resized_annotation = self.resize_annotation(target, orig_size, get_image_size(resized_image))
+                    resized_images.append(resized_image)
+                    resized_annotations.append(resized_annotation)
+                images = resized_images
+                annotations = resized_annotations
+                del resized_images, resized_annotations
+            else:
+                images = [self.resize(image, size=size, resample=resample) for image in images]
+
+        if do_rescale:
+            images = [self.rescale(image, rescale_factor) for image in images]
+
+        if do_normalize:
+            images = [self.normalize(image, image_mean, image_std) for image in images]
+            if annotations is not None:
+                annotations = [
+                    self.normalize_annotation(annotation, get_image_size(image))
+                    for annotation, image in zip(annotations, images)
+                ]
+
+        if do_pad:
+            # Pads images and returns their mask: {'pixel_values': ..., 'pixel_mask': ...}
+            data = self.pad(images, return_pixel_mask=True, data_format=data_format)
+        else:
+            images = [to_channel_dimension_format(image, data_format) for image in images]
+            data = {"pixel_values": images}
+
+        encoded_inputs = BatchFeature(data=data, tensor_type=return_tensors)
+        if annotations is not None:
+            encoded_inputs["labels"] = [
+                BatchFeature(annotation, tensor_type=return_tensors) for annotation in annotations
+            ]
+
+        return encoded_inputs
+
+    def post_process_object_detection(
+        self, outputs, threshold: float = 0.5, target_sizes: Union[TensorType, List[Tuple]] = None
+    ):
+        """
+        Converts the output of [`DetaForObjectDetection`] into the format expected by the COCO api. Only supports
+        PyTorch.
+
+        Args:
+            outputs ([`DetrObjectDetectionOutput`]):
+                Raw outputs of the model.
+            threshold (`float`, *optional*):
```

Comment:
We can't just add NMS support to all post_process_object_detection methods I'm afraid, since for that one uses Torchvision's NMS op which is written in C and much faster than plain python. Also our existing models don't need NMS, DETA is actually the first one that needs it

Reply:
NMS is a quite common to post-processing method for object detection and useful for models with noisy bounding box proposals. Given two bounding boxes with high prediction scores, they might be detecting the same object instance and NMS eliminates the redundant boxes.

I think it'd be nice to have an option to perform NMS but I'm fine with not adding it as well. 
CC'ing @amyeroberts and @sgugger

Reply:
> since for that one uses Torchvision's NMS op which is written in C and much faster than plain python

@NielsRogge I'm not sure I completely follow - is the issue here speed or the use of torchvision? 

In general, I agree with @alaradirik, NMS is common enough that it's something I think we want to support, even if it's only for our PyTorch models at the moment.

Reply:
Ok we can add NMS as an option, this way the API of all `post_process_object_detection` methods will be the same. I assume we'll want to leverage torchvision for this

Reply:
Yes, it makes sense to leverage torchvision for this